src/main/java/jnr/ffi/Struct.java

```diff
@@ -101,13 +103,12 @@ protected final int addField(int sizeBits, int alignBits, Offset offset) {
         }
 
         protected final int addField(int sizeBits, int alignBits) {
-            final int off = resetIndex ? 0 : align(this.size, alignBits >> 3);
-            this.size = Math.max(this.size, off + (sizeBits >> 3));
-            this.minAlign = Math.max(this.minAlign, alignBits >> 3);
-            return off;
+            final int alignment = this.alignment.intValue() > 0 ? Math.min(this.alignment.intValue(), (alignBits >> 3)) : (alignBits >> 3);
```

Comment:
@goto1134 is not this test case already tested in this [test case](https://github.com/jnr/jnr-ffi/pull/46/files#diff-bd5234f278cf4e3ece9bfc282ac64224R164)?